iOSClient/Diagnostics/NCCapabilitiesView.swift

```diff
@@ -0,0 +1,236 @@
+            }
+        }
+
+        updateCapabilities(account: account)
+    }
+
+    func updateCapabilities(account: String) {
+
+        var available: Bool = false
+
+        capabililies.removeAll()
+        json = ""
+
+        // FILE SHARING
+        available = NCManageDatabase.shared.getCapabilitiesServerBool(account: account, elements: NCElementsJSON.shared.capabilitiesFileSharingApiEnabled, exists: false)
+        capabililies.append(Capability(text: "File sharing", image: UIImage(named: "share")!.resizeImage(size: CGSize(width: 25, height: 25))!, available: available))
+
+        // EXTERNAL SITE
+        available = NCManageDatabase.shared.getCapabilitiesServerBool(account: account, elements: NCElementsJSON.shared.capabilitiesExternalSites, exists: true)
+        capabililies.append(Capability(text: "External site", image: UIImage(systemName: "network")!, available: available))
+
+        // E2EE
+        available = NCManageDatabase.shared.getCapabilitiesServerBool(account: account, elements: NCElementsJSON.shared.capabilitiesE2EEEnabled, exists: false)
+        capabililies.append(Capability(text: "End-to-End Encryption", image: UIImage(systemName: "lock")!, available: available))
+
+        // ACTIVITY
+        if NCManageDatabase.shared.getCapabilitiesServerArray(account: account, elements: NCElementsJSON.shared.capabilitiesActivity) == nil {
+            available = false
+        } else {
+            available = true
+        }
+        capabililies.append(Capability(text: "Activity", image: UIImage(systemName: "bolt")!, available: available))
+
+        // NOTIFICATION
+        if NCManageDatabase.shared.getCapabilitiesServerArray(account: account, elements: NCElementsJSON.shared.capabilitiesNotification) == nil {
+            available = false
+        } else {
+            available = true
+        }
+        capabililies.append(Capability(text: "Notification", image: UIImage(systemName: "bell")!, available: available))
+
+        // DELETE FILES
+        available = NCManageDatabase.shared.getCapabilitiesServerBool(account: account, elements: NCElementsJSON.shared.capabilitiesFilesUndelete, exists: false)
+        capabililies.append(Capability(text: "Deleted files", image: UIImage(systemName: "trash")!, available: available))
+
+        // TEXT - ONLYOFFICE
+        var textEditor = false
+        var onlyofficeEditors = false
+        if let editors = NCManageDatabase.shared.getDirectEditingEditors(account: account) {
+            for editor in editors {
+                if editor.editor == NCGlobal.shared.editorText {
+                    textEditor = true
+                } else if editor.editor == NCGlobal.shared.editorOnlyoffice {
+                    onlyofficeEditors = true
+                }
+            }
+        }
+        capabililies.append(Capability(text: "Text", image: UIImage(named: "text")!.resizeImage(size: CGSize(width: 25, height: 25))!, available: textEditor))
```

Comment:
Lots of force unwrapping here. Can we use `guard`?

Reply:
You are mostly right, but there are edge cases where you might rename an asset or remove it and the whole app crashes because of a random image. I still think guarding is a good idea here but up to you.